presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -428,30 +411,28 @@ protected Boolean visitIdentifier(Identifier node, Void context)
             if (analysis.getLambdaArgumentReferences().containsKey(node)) {
                 return true;
             }
-            return isField(node, QualifiedName.of(node.getName()));
+            return isColumnReferenceUseCorrect(node);
         }
 
         @Override
         protected Boolean visitDereferenceExpression(DereferenceExpression node, Void context)
         {
-            if (analysis.getColumnReferences().contains(node)) {
-                return isField(node, DereferenceExpression.getQualifiedName(node));
+            if (columnReferences.containsKey(node)) {
+                return isColumnReferenceUseCorrect(node);
             }
 
             // Allow SELECT col1.f1 FROM table1 GROUP BY col1
             return process(node.getBase(), context);
         }
 
-        private boolean isField(Expression node, QualifiedName qualifiedName)
+        private boolean isColumnReferenceUseCorrect(Expression node)
```

Comment:
Yeah, what this class does is "check whether an expression is constant with respect to the group" (see class javadoc).

The problem I have with this method's name is that it's too ambiguous. Maybe name it something like "isGroupingKey".